inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/MysqlToRocksTest.java

```diff
@@ -0,0 +1,217 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sort.tests;
+
+import org.apache.inlong.sort.tests.utils.FlinkContainerTestEnv;
+import org.apache.inlong.sort.tests.utils.JdbcProxy;
+import org.apache.inlong.sort.tests.utils.MySqlContainer;
+import org.apache.inlong.sort.tests.utils.StarRocksContainer;
+import org.apache.inlong.sort.tests.utils.TestUtils;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.MountableFile;
+
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * End-to-end tests for sort-connector-postgres-cdc-v1.15 uber jar.
+ * Test flink sql Postgres cdc to StarRocks
+ */
+public class MysqlToRocksTest extends FlinkContainerTestEnv {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PostgresToStarRocksTest.class);
+
+    private static final Path mysqlJar = TestUtils.getResource("sort-connector-mysql-cdc.jar");
+    private static final Path jdbcJar = TestUtils.getResource("sort-connector-starrocks.jar");
+    private static final Path mysqlJdbcJar = TestUtils.getResource("mysql-driver.jar");
+
+    private static final Logger STAR_ROCKS_LOG = LoggerFactory.getLogger(StarRocksContainer.class);
+
+    private static final String sqlFile;
+
+    // ----------------------------------------------------------------------------------------
+    // StarRocks Variables
+    // ----------------------------------------------------------------------------------------
+    private static final String INTER_CONTAINER_STAR_ROCKS_ALIAS = "starrocks";
+    private static final String NEW_STARROCKS_REPOSITORY = "inlong-starrocks";
+    private static final String NEW_STARROCKS_TAG = "latest";
+    private static final String STAR_ROCKS_IMAGE_NAME = "starrocks/allin1-ubi:3.0.4";
+
+    static {
+        try {
+            sqlFile =
+                    Paths.get(PostgresToStarRocksTest.class.getResource("/flinkSql/mysql_test.sql").toURI()).toString();
+            buildStarRocksImage();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String getNewStarRocksImageName() {
+        return NEW_STARROCKS_REPOSITORY + ":" + NEW_STARROCKS_TAG;
+    }
+
+    public static void buildStarRocksImage() {
```

Comment:
duplicate with test code in postgreToStarrocks

Reply:
make a base class and reuse this logic ?

Reply:
Yes it's one approach or you can create a class just for the initialization for starrocks and reuse it